DAQ/src/StrawDigisFromArtdaqFragments_module.cc

```diff
@@ -0,0 +1,437 @@
+#include "artdaq-core-mu2e/Overlays/DTC_Packets/DTC_RocDataHeaderPacket.h"
+
+#include "Offline/DataProducts/inc/TrkTypes.hh"
+#include "Offline/RecoDataProducts/inc/IntensityInfoTrackerHits.hh"
+// #include "Offline/RecoDataProducts/inc/ProtonBunchTime.hh"
+#include "Offline/RecoDataProducts/inc/StrawDigi.hh"
+
+#include <artdaq-core/Data/Fragment.hh>
+#include <artdaq-core/Data/ContainerFragment.hh>
+
+#include <iostream>
+
+#include <string>
+
+#include <memory>
+
+// #define TRACEMF_USE_VERBATIM 1
+
+// #include "TRACE/tracemf.h"
+// #define TRACE_NAME "StrawDigisFromArtdaqFragments"
+
+
+namespace art {
+  class StrawDigisFromArtdaqFragments;
+}
+// ======================================================================
+
+class art::StrawDigisFromArtdaqFragments : public EDProducer {
+
+public:
+
+  struct Config {
+    fhicl::Atom<int> diagLevel    {fhicl::Name("diagLevel"    ), fhicl::Comment("diagnostic severity level, default = 0" ), 0};
+    fhicl::Atom<int> debugLevel   {fhicl::Name("debugLevel"   ), fhicl::Comment("debug level, default = 0"               ), 0};
+    fhicl::Atom<int> saveWaveforms{fhicl::Name("saveWaveforms"), fhicl::Comment("save StrawDigiADCWaveforms, default = 1"), 1};
+  };
+
+  // --- C'tor/d'tor:
+  explicit StrawDigisFromArtdaqFragments(const art::EDProducer::Table<Config>& config);
+  virtual ~StrawDigisFromArtdaqFragments() {}
+
+  int          panelID(uint16_t ChannelID, int DtcID, int LinkID);
+
+  void         print_(const std::string&  Message, int DiagLevel = -1,
+                      const std::source_location& location = std::source_location::current());
+
+  void         print_fragment(const artdaq::Fragment* Frag);
+
+    // --- overloaded functions of the art producer
+  virtual void produce (art::Event& ArtEvent) override;
+  virtual void beginRun(art::Run&   ArtRun  ) override;
+
+private:
+                                        // talk-to parameters
+  int       diagLevel_    ;
+  int       debugLevel_   ;
+  int       saveWaveforms_;
+                                        // the rest
+  int       nADCPackets_{-1};           // N(ADC packets per hit)
+  int       nSamples_   {-1};           // N(ADC samples per hit)
+  int       np_per_hit_ {-1};           // N(data packets per hits)
+
+  int       panel_map_[36][6];          // panel_map_[idtc][ilink] = panel MN number
+
+  const art::Event* event_;
+
+};
+
+// ======================================================================
+art::StrawDigisFromArtdaqFragments::StrawDigisFromArtdaqFragments(const art::EDProducer::Table<Config>& config) :
+    art::EDProducer{config},
+    diagLevel_    (config().diagLevel    ()),
+    debugLevel_   (config().debugLevel   ()),
+    saveWaveforms_(config().saveWaveforms())
+{
+  produces<mu2e::StrawDigiCollection>();
+  if (saveWaveforms_) produces<mu2e::StrawDigiADCWaveformCollection>();
+
+  produces<mu2e::IntensityInfoTrackerHits>();
+  // FIXME!
+  //  produces<mu2e::ProtonBunchTime>();
```

Comment:
Is this comment still relevant? ProtonBunchTime is a high-level object coming from analyzing (potentially) multiple data products and applying physics corrections, why would it be produced by a low-level module such as this?